### ISSUE TITLE
Random passthrough issues on nvidia shield tv

### ISSUE THREAD

**Issue by tstack77** [koying/SPMC#748]
Raw passthrough is enabled and mostly functioning properly. About 20% of the time that I start a movie/show the audio stream fails to play from my receiver (yamaha rx-v677). The receiver's display will change and show the correct stream type when working, but when the stream fails it never changes from the 'PCM' tag.

I am able to correct the problem by disabling/re-enabling passthrough in the media's audio properties. I have tried to add a pause delay to the playback but that hasn't helped. Any ideas?

EDIT: Connection is through HDMI. Issue found with DTS, DTS-HD, and AC-3. I will check the log later tonight when I get home.

**Comment by joergbattermann** [comment]
Yep I can confirm that behaviour.. same for my Denon AVR-X2000 - it appears to have started after upgrading to the Shield 5.0.x / Android 7.x version.. but I can't say for sure. Restarting the entire device does fix it, but that's a bit annoying.

**Comment by randomginger** [comment]
I can also confirm on Yamaha RXV 681 that AC3 works for half a second then sound drops off completely. I have to go to sound setting, disable, then enable pass through to get sound working again. I am connected via HDMI and also think this happened since last shield update.

**Comment by randomginger** [comment]
I just found this to resolve the problem.

https://github.com/koying/SPMC/issues/743

I have to set my delay to 4 seconds.

**Comment by joergbattermann** [comment]
thanks @randomginger - that did the trick for me, too 👍

**Comment by tstack77** [comment]
Thanks randomginger. I had a pause already in place to solve a lipsyncing issue, turns out .5 seconds isn't enough

**Comment by sudo-tee** [comment]
I have no sound with DTS and DD files in SPMC since the update. I tried 3 second delay and it does nothing. I will try again with 4.

What are your setting in Android > Audio & Video > Advanced

**Comment by randomginger** [comment]
Yes, I have Shield and settings are -

IEC pass through
7.1 channels
Optimized
Stereo upmix off
Original volume on down mix off
Boost centre 0db
Resample high
Keep audio alive 1 minute
Enable DSP processing off
GUI sounds only when playback stopped
Enable pass through on
Passthrough output IEC pass through
All sound settings set to on.

In playback settings-
Adjust display refresh rate always
Pause during refresh rate change 4secs
Sync to display off

Hope it helps.

On 10 Feb 2017 12:27, inkubux <notifications@github.com> wrote:

I have no sound with DTS and DD files in SPMC since the update. I tried 3 second delay and it does nothing. I will try again with 4.

What are your setting in Android > Audio & Video > Advanced

**Comment by Yomark1** [comment]
Have the same thing with a Onkyo TX-SR656. Setting the delay fixes it for the most part, but sometimes SPMC does not seems to pause at all.
A friend of mine has an old Denon receiver , and also has these problems. Seems like a very widespread issue with the latest Nvidia update.

**Comment by imbaimbaimba** [comment]
I got the same issue with a Denon X3300 AVR since the latest Shiel update. The delay setting does however not seem to fix it for me :( 

When I start something, about 1/5 times it correctly passes through whatever sound format I am playing. The rest of the time, there is no sound, and my receiver just says "Stereo". If I press the stop button, and start it again, usually it works. Sometimes I have to stop-start upwards 10 times though.

Any news on this?

**Comment by lovemyram4x4** [comment]
I've been having pass through issue with my 2017 model(probably my updated 2015 model would also with update but I'm not currently using pass through).

I initially noticed I had the issue with output and pass through both set to andriod RAW but if I changed output to either PCM or EIC I didn't have the pass through issues.

The other night I started having a 6.7sec lip sync issue even with setting a 4 sec pause on refresh change(I never needed the pause on my Shield TV as the screen only went black for a split sec-when this problem surfaced it went black for a couple of seconds). Ironically putting audio output back to RAW it went back to no issues and no black screen on refresh change.

No the no audio issue has returned again(BTW AVR says decoder off), I changed output to PCM and audio is OK again but I get 1 or 2 sec of black screen of refresh change..

As other mentioned if having the issue you can often turn off pass through then turn it back on and get sound(when I was having the sync issue it didn't help with that).
